### PR TITLE
[TST][DOC] Refactor currency conversion

### PR DESCRIPTION
**Pull request by dendrondal**
# PR Description

Please describe the changes proposed in the pull request:

-Moved currency_column_to_numeric from funcitons.py to finance, as I can't see this being used outside of finance applications 
- Moved documentation and added tests accordingly.
-

**This PR resolves N/A **

# PR Checklist

Please ensure that you have done the following:

1. [X] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [X] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [X] Add a line to `CHANGELOG.rst` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [X] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build

Once done, please check off the check-box above.

If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [X] Ensure that you have added tests.
- [X] Run all tests (`$ pytest .`) locally on your machine.
    - [X] Check to ensure that test coverage covers the lines of code that you have added.
    - [X] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [X] Build the docs locally.
- [X] View the docs to check that it renders correctly.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ new version (on deck)
 - [DOC] Fix contributions hyperlinks in ``AUTHORS.rst`` and contributions by @hectormz
 - [INF] Add ``pre-commit`` hooks to repository by @ericmjl
 - [DOC] Fix formatting code in ``CONTRIBUTING.rst`` by @hectormz
+- [TST] Moved currency_column_to_numeric from functions to finance, added tests
```
Comment (ericmjl) [inline CHANGELOG.rst]:
```suggestion
- [TST] Moved currency_column_to_numeric from functions to finance, added tests by @dendrondal 
```

**Comment by hectormz** [comment]
@ericmjl @dendrondal Should we use `refactored_function` as in #556 to account for moving a function to another module?

**Comment by dendrondal** [comment]
I like that pattern! I think it would work well here.

**Comment by ericmjl** [comment]
@dendrondal let's add the `@refactored_function` decorator then. Thanks for the catch, @hectormz!

File: janitor/finance.py [hunk 2]
```diff
@@ -395,6 +402,237 @@ def _convert_currency(
     return rate
 
 
+@pf.register_dataframe_method
+@deprecated_alias(col_name="column_name", type="cleaning_style")
+@refactored_function(
```
Comment (hectormz) [inline janitor/finance.py]: Hi @dendrondal, you don't need to import and implement `@refactored_function(...)` here.

It should go in `__init__.py`, and you can follow the pattern that `get_features_targets` uses:

```python
from .finance import currency_column_to_numeric as _currency_column_to_numeric

@refactored_function(
    "currency_column_to_numeric() has moved. Please use finance.currency_column_to_numeric()."
)
def currency_column_to_numeric(*args, **kwargs):
    return _currency_column_to_numeric(*args, **kwargs)

```

If I recall, that should be the last thing to push this PR through!

Comment (ericmjl) [reply janitor/finance.py]: @hectormz, thank you for conducting the review! I'm also learning from you how this pattern can be used :smile:. 

@dendrondal - thanks for handling this! Once you're done with that, don't worry about the conflict on CHANGELOG. I will take care of that through the web editor.

Comment (hectormz) [reply janitor/finance.py]: 👍 Perhaps I should write a small section in `CONTRIBUTING.rst` on how and why to refactor functions and argument names with the wrappers we've implemented. Maybe even a list item in the PR template whether or not someone has moved a function/renamed arguments. Might be overkill though...

Comment (dendrondal) [reply janitor/finance.py]: This should do the trick! I think adding a subsection to CONTRIBUTING.rst is a good idea, but it seems like such a rare occurrence that I don't know if it needs to be in the PR template.

**Comment by ericmjl** [comment]
@dendrondal, let me know when you're done with this, then I'll go ahead and resolve the conflicts via the web editor and merge.

**Comment by dendrondal** [comment]
That should be everything!

**Comment by hectormz** [comment]
@ericmjl @dendrondal where did this leave off? Do we need to update from `dev` and re-start the checks? Or did the failed checks have other issues?

**Comment by ericmjl** [comment]
@hectormz yes, I think a updating from `dev` and restarting is the right thing to do. There will be conflicts in the files, though, so I suspect it might be a substantial number of changes needed here.

In the interest of keeping the PR pipeline clean, I'm going to close the PR. @dendrondal or anybody else, if you wish to pick it up again, please feel free to study the PR for inspiration - I will humbly suggest branching off latest `dev` for a new PR.

**Pull request closed without merging**